hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/OMMetadataManager.java

```diff
@@ -116,6 +116,22 @@ public interface OMMetadataManager extends DBStoreHAManager {
    */
   String getBucketKey(String volume, String bucket);
 
+  /**
+   * Given a volume and bucket, return the corresponding DB key prefix.
+   *
+   * @param volume - User name
+   * @param bucket - Bucket name
+   */
+  String getBucketKeyPrefix(String volume, String bucket);
```

Comment:
We have similar functions in [SnapshotUtils](https://github.com/apache/ozone/blob/7a452ca2e84520c2fb15356fde38857cacb758c5/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotUtils.java#L202-L244). Can you please move them from here?

Reply:
We should be ideally having all such methods in OmMetadataManager not in SnapshotUtils. We shouldn't be having the logic of key layout in some other Util function. Our snapshotUtils has been written very specific to OmMetadataManagerImpl which is just bad design in my opinion.

Reply:
Snapshot code shouldn't be hitting metadata manager layer. All calls should be made to KeyManager & keyManager should hit metadata layer. From what I understand, this was the intent of creation of these interface initially. We ended up mixing all of this logic everywhere by directly accessing tables from all the places.

Reply:
We should such similar things done on the Snapshot diff flow & SSTFiltering service as well. We can create a refactoring jira for it. Could be a good newbie jira. Would be a good opportunity for someone who wants to understand snapshot flow.

Reply:
The trailing slash is mainly used by snapshots as a key prefix. Also, [the comment](https://github.com/apache/ozone/blob/7a452ca2e84520c2fb15356fde38857cacb758c5/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotUtils.java#L208) says that it is meant to be used in snapshots and should be moved out if used out snapshots. Since this will be used at other places, it is a good time to move them to `OmMetadataManager`.

I'm fine if you create a Jira for it. My point was not to have duplicate functions sitting in multiple places.